daemon/api.go

```diff
@@ -661,6 +661,7 @@ func postSnap(c *Command, r *http.Request, user *auth.UserState) Response {
 
 	chg := newChange(state, inst.Action+"-snap", msg, tsets)
 	chg.Set("snap-names", []string{inst.snap})
+	chg.Set("api-data", map[string]string{"snap-name": inst.snap})
```

Comment:
Yeah, this is being done for the "for" functionality in snap changes, and we've just found a wart which we should fix. snap-names should be looked at inside api-data so it's exposed to the API, and we should either have snap-names or snap-name on each change, or perhaps always snap-names if we want it to be more obvious to consume. Having both with different logic (single vs. list) and in different places feels very undesirable.